### DISCUSSION TITLE
Contribute a welcome patch to base/utils/main C code

### DISCUSSION THREAD

**Discussion by hturner** [r-devel/r-project-sprint#23] [Ideas]
I have been reviewing reports on Bugzilla and have found several where a member of R Core has indicated a patch would be welcome, either by their comments or by adding the HELPWANTED keyword. These bugs are good candidates for contributors to take on at the sprint, though they generally require advanced knowledge of R and/or C. In some cases, the work may extend beyond the sprint, but the sprint can provide an opportunity to get advice from R Core/other experienced contributors.

| Bug ID                                                    | Level        | Topic/Domain    | Summary                                                                    | Language/other |
|-----------------------------------------------------------|--------------|-----------------|----------------------------------------------------------------------------|----------------|
| [18255](https://bugs.r-project.org/show_bug.cgi?id=18255) | Intermediate | packages        | requireNamespace doesn't respect min version when namespace already loaded | R              |
| [17871](https://bugs.r-project.org/show_bug.cgi?id=17871) | Intermediate | utils           | `utils::tar` can produce invalid tarballs w/ long usernames/groupnames     | R              |
| [14354](https://bugs.r-project.org/show_bug.cgi?id=14354) | Advanced     | packages        | mark non-standard evaluation in local scope to be skipped by codetools     | R              |
| [16094](https://bugs.r-project.org/show_bug.cgi?id=16094) | Advanced     | packages        | respect package version restrictions when using `::`                       | R              |
| [16311](https://bugs.r-project.org/show_bug.cgi?id=16311) | Advanced     | base            | set environment variables from env argument                                | C, Windows     |
| [16752](https://bugs.r-project.org/show_bug.cgi?id=16752) | Advanced     | base            | printing complex numbers in R                                              | C              |
| [16856](https://bugs.r-project.org/show_bug.cgi?id=16856) | Advanced     | base            | addition of millisecond adds a second                                      | C              |
| [16034](https://bugs.r-project.org/show_bug.cgi?id=16034) | Advanced     | base            | iterative unserialization for pairlists                                    | C              |
| [17377](https://bugs.r-project.org/show_bug.cgi?id=17377) | Advanced     | utils           | inconsistent result of getSrcLocation for an S4 method that uses `...`     | R              |
| [18249](https://bugs.r-project.org/show_bug.cgi?id=18249) | Advanced     | src/main C code | following Unicode TR31 specification in R                                  | C              |

Additions or corrections to this list are welcome - note I have not checked if the bugs are current in R-devel.

R Core members commenting on original reports: @mmaechler, @ltierney, @bastistician, @s-u.

**Comment by mmaechler** [comment]
That's really nice -- thank you @hturner!  Very good you also picked some "oldies".
I think that indeed quite a few additions should be possible!
